### PR TITLE
add rerankK to GraphSearcher::search, and worstApproximateScoreInTopK to SearchResult

### PR DESCRIPTION
**Pull request by jbellis**

### CODE REVIEW

File: jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphSearcher.java [hunk 1]
```diff
@@ -115,6 +114,7 @@ public GraphSearcher build() {
      * @param scoreProvider   provides functions to return the similarity of a given node to the query vector
      * @param topK            the number of results to look for. With threshold=0, the search will continue until at least
      *                        `topK` results have been found, or until the entire graph has been searched.
+     * @param rerankK         the number of (approximately-scoared) results to rerank before returning the best `topK`.
```
Comment (jkni) [inline jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphSearcher.java]: s/scoared/scored

File: jvector-examples/src/main/java/io/github/jbellis/jvector/example/Grid.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ static void runAll(DataSet ds,
                 }
             }
         } finally {
-            Files.delete(testDirectory);
+//            Files.delete(testDirectory);
```
Comment (jkni) [inline jvector-examples/src/main/java/io/github/jbellis/jvector/example/Grid.java]: Uncomment this?

File: jvector-base/src/main/java/io/github/jbellis/jvector/graph/NodeQueue.java [hunk 3]
```diff
@@ -139,17 +138,64 @@ public int[] nodesCopy() {
...
-                                    .mapToInt(this::decodeNodeId).toArray();
-        var scores = reranker.similarityTo(ids);
-        var nodeScores = new SearchResult.NodeScore[ids.length];
+    /**
+     * Rerank results and return the worst approximate score that made it into the topK.
+     * The topK results will be placed into `reranked`, and the remainder into `unused`.
+     */
+    public float rerank(int topK, ScoreFunction.Reranker reranker, float rerankFloor, NodeQueue reranked, NodesUnsorted unused) {
+        // Rescore the nodes whose approximate score meets the floor.  Nodes that do not will be marked as -1
+        // We will index into the original heap to retrieve corresponding approximate scores later,
```
Comment (jkni) [inline jvector-base/src/main/java/io/github/jbellis/jvector/graph/NodeQueue.java]: Is this comment still true? I don't see us indexing into the original heap

Comment (jbellis) [reply jvector-base/src/main/java/io/github/jbellis/jvector/graph/NodeQueue.java]: thanks, there's one place where we can do this (done in latest commit) but mostly this is misleading now I think

**Pull request merged**